### ISSUE TITLE
Agent.reconnect process stuck forever for the agents not linked to any host

### ISSUE THREAD

**Issue by alena1108** [rancher/rancher#5349]
Seen on some users setup:

mysql> select resource_id, start_time, exit_reason from process_instance where process_name='agent.reconnect' and end_time is null and resource_id in (select id from agent where id not in (select agent_id from host) and removed is null and state='reconnecting');
+-------------+---------------------+-------------+
| resource_id | start_time          | exit_reason |
+-------------+---------------------+-------------+
| 50          | 2016-01-28 03:15:43 | TIMEOUT     |
| 108         | 2016-02-16 03:03:54 | TIMEOUT     |
| 109         | 2016-02-16 03:04:54 | TIMEOUT     |
| 132         | 2016-02-16 04:04:39 | TIMEOUT     |
| 133         | 2016-02-16 04:04:59 | TIMEOUT     |
| 277         | 2016-02-22 08:40:45 | TIMEOUT     |
| 278         | 2016-02-22 08:40:55 | TIMEOUT     |
| 4047        | 2016-04-01 02:56:20 | TIMEOUT     |
| 4050        | 2016-04-01 03:02:15 | TIMEOUT     |
| 4055        | 2016-04-01 03:43:09 | TIMEOUT     |
| 7351        | 2016-04-06 02:27:38 | TIMEOUT     |
| 7385        | 2016-04-06 03:25:18 | TIMEOUT     |
| 7868        | 2016-04-26 09:47:48 | TIMEOUT     |
| 11338       | 2016-05-16 06:32:23 | TIMEOUT     |
| 11979       | 2016-05-17 03:15:04 | TIMEOUT     |
| 13840       | 2016-05-28 03:57:02 | TIMEOUT     |
| 13953       | 2016-05-30 04:35:23 | TIMEOUT     |
| 14077       | 2016-05-31 09:18:54 | TIMEOUT     |
+-------------+---------------------+-------------+

these processes are stuck in execution loop forever, and as they take execution precedence based on the date, all the subsequent processes get slowed down.

**Comment by schuylr** [comment]
For people looking for a workaround for now, go to Admin > Processes, select the Running tab

Look for `agent.reconnect` processes (make sure all of your hosts are okay before touching any of these):

![screen shot 2016-09-15 at 7 23 41 pm](https://cloud.githubusercontent.com/assets/12628715/18570685/1bb006d2-7b7a-11e6-975a-c0f423d5d820.png)

Click on the `agent:1a123` link in the 3rd column to open the agent in the API.

Click on "deactivate", click "Show Request", then click "Send Request", and if you're feeling bold, click "Reload" (may need to reload the page a few times) and click "remove"/"Show Request"/"Send Request"

This will leave the agents in a removed state scheduled for purge later.

@vincent99 aptly found that I was using `restart: always` in my cloud config for the `register` agent, which is only supposed to run once. It caused things like #5950 and #5796 because the `agent.reconnect` processes were hogging the queue with all of my hosts that were creating multiple agents with different IDs. @cjellick was very helpful and figured out this second part with me.

Moral of the story: don't set the registration agent to `--restart=always`. Thanks all!

**Comment by demarant** [comment]
we are experiencing same issues.  Our rancher server is full of stuck processes.Check the list of processes we got below... 

restarting rancher only helps temporarily. the stuck processes comes back and entire rancher is basically very slow and unusable. we are experience service disruptions for our services in<production environments because of this bug :( 

I will try the solution from @schuylr 

we expect rancher to cleanup itself or provide means to admin to cleanup these processes.

![rancher-processes](https://cloud.githubusercontent.com/assets/320440/19469053/ef247e2c-9519-11e6-85b9-7688434f035c.png)

**Comment by demarant** [comment]
@schuylr 
thank you! I managed to remove all the agent.reconnect that were stuck for months with the method via the API... that worked for us. Now the server is able to breath again. 
it doesn't work for other processes though, like volume.activate ... but i think i'll create another issue for that.

**Comment by ibuildthecloud** [comment]
After an agent is in reconnecting for 10 minutes it will go to `disconnected`.  Not until the agent connects again will the agent leave the `disconnected` state.  `disconnected` is a resting state so there is no process associated with it.

**Comment by deniseschannon** [comment]
With the changes in 1.2 and no longer having agents remaining in reconnecting, there should be no open processes for reconnecting. If this is seen with new agents after 1.2, please re-open.
